torchx/schedulers/kueue_job_scheduler.py

```diff
@@ -0,0 +1,809 @@
+        except Exception as e:
+            logger.exception("Unable to retrieve job name, got exception", e)
+            return None
+
+    def _get_active_context(self) -> Dict[str, Any]:
+        from kubernetes import config
+
+        contexts, active_context = config.list_kube_config_contexts()
+        return active_context
+
+    def schedule(self, dryrun_info: AppDryRunInfo[KueueJob]) -> str:
+        from kubernetes.client.rest import ApiException
+
+        cfg = dryrun_info._cfg
+        assert cfg is not None, f"{dryrun_info} missing cfg"
+        namespace = cfg.get("namespace") or "default"
+
+        images_to_push = dryrun_info.request.images_to_push
+        self.push_images(images_to_push)
+
+        resource = dryrun_info.request.resource
+        try:
+            resp = self._batchv1_api().create_namespaced_job(
+                namespace=namespace,
+                body=resource,
+            )
+
+        except ApiException as e:
+            if e.status == 409 and e.reason == "Conflict":
+                job_name = self._get_job_name_from_exception(e)
+                raise ValueError(
+                    f"Job `{job_name}` already exists. This seems like a transient exception, try resubmitting job"
+                ) from e
+            else:
+                raise
+
+        return f'{namespace}:{resp.metadata.name}'
+
+    def _submit_dryrun(
+        self, app: AppDef, cfg: KueueJobOpts
+    ) -> AppDryRunInfo[KueueJob]:
+        # map any local images to the remote image
+        images_to_push = self.dryrun_push_images(app, cast(Mapping[str, CfgVal], cfg))
+
+        service_account = cfg.get("service_account")
+        assert service_account is None or isinstance(
+            service_account, str
+        ), "service_account must be a str"
+
+        local_kueue = cfg.get("local_kueue")
+        assert local_kueue is None or isinstance(
+            local_kueue, str
+        ), "local_kueue must be a str"
+        
+        kueue_priority_class = cfg.get("kueue_priority_class")
+        assert kueue_priority_class is None or isinstance(
+            kueue_priority_class, str
+        ), "kueue_priority_class must be a str"
+        
+        resource = app_to_resource(app, service_account, local_kueue,  kueue_priority_class)
+        req = KueueJob(
+            resource=resource,
+            images_to_push=images_to_push,
+        )
+        return AppDryRunInfo(req, repr)
+
+    def _validate(self, app: AppDef, scheduler: str) -> None:
+        # Skip validation step
+        pass
+
+    def _cancel_existing(self, app_id: str) -> None:
+        from kubernetes import client
+        namespace, name = app_id.split(":")
+
+        self._batchv1_api().delete_namespaced_job(
+            namespace=namespace,
+            name=name,
+            body=client.V1DeleteOptions(
+                    propagation_policy='Foreground',
+                    grace_period_seconds=5))
```

Comment:
What's the reason behind not just using the default `grace_period`?

Reply:
I was testing something and forgot to remove the grace periods that can go :+1: